mlir/utils/jenkins/Jenkinsfile

```diff
@@ -1057,5 +1058,68 @@ pipeline {
+                    // Note the %m to avoid issues with threads and dynamic libraries.
+                    LLVM_PROFILE_FILE="${WORKSPACE}/build/%m-%p.profraw"
+                }
+                stages {
+                    stage ("body") {
+                        steps {
+                            // Build with profiling on, and just code-generation tests.
+                            sh 'rm -f build/CMakeCache.txt'
+                            sh 'rm -f build/*.profraw'
+                            buildProject('check-rocmlir-build-only',
+                                         '-DBUILD_FAT_LIBROCKCOMPILER=ON -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS_DEBUG="-O0 -fprofile-instr-generate -fcoverage-mapping"')
+                            dir ('build') {
+                                // Run tests.
+                                sh 'ninja check-rocmlir'
+                                // Profile processing.
+                                sh '/opt/rocm/llvm/bin/llvm-profdata merge -sparse ./*.profraw -o ./coverage.profdata'
+                                sh '/opt/rocm/llvm/bin/llvm-cov report --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project > ./coverage.report'
+                                sh 'cat ./coverage.report'
+                                sh '/opt/rocm/llvm/bin/llvm-cov show --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project -Xdemangler=/opt/rocm/llvm/bin/llvm-cxxfilt > ./coverage.info'
+                                sh '/opt/rocm/llvm/bin/llvm-cov show --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project -Xdemangler=/opt/rocm/llvm/bin/llvm-cxxfilt --format=html > ./coverage.html'
+                                // Upload to codecov.
+                                withCredentials([string(credentialsId: 'codecov-token-rocmlir', variable: 'CODECOV_TOKEN')]) {
+                                    sh '''
+                                       curl -Os https://uploader.codecov.io/latest/linux/codecov && chmod +x ./codecov
+                                       proxy_opt=""
+                                       if [ "${CHIP}" = "gfx90a" ]; then
+                                           proxy_opt='-U http://172.23.0.23:3128'
+                                       fi
+                                       ./codecov -t ${CODECOV_TOKEN} --flags "${CHIP}" -f ./coverage.info ${proxy_opt}
+                                       '''
+                                }
+                            }
+                            archiveArtifacts artifacts: 'build/coverage.report, build/coverage.info, build/coverage.html', onlyIfSuccessful: true
```

Comment:
Archiving mostly to save the html so we at least have something to look at.  Some projects do further processing to make coverage diffs or to detect decreased coverage after pull requests, both of which would be future projects for us.